src/crimson/osd/object_context_loader.cc

```diff
@@ -116,6 +127,15 @@ using crimson::common::local_conf;
         }
         obc->set_head_state(std::move(md->os),
                             std::move(md->ssc));
+	for (auto &[src, winfo] : obc->obs.oi.watchers) {
+	  auto [it, emplaced] = obc->watchers.emplace(src, nullptr);
+	  assert(emplaced);
+	  logger().debug("adding watch for obj {}, client {}",
+	    obc->get_oid(), src.second);
+	  it->second = crimson::osd::Watch::create(
+	    obc, winfo, src.second, std::move(pg));
+	  it->second->disconnect();
```

Comment:
this is a little bit confusing and each `src` moves the `pg` away, this leaves next watcher an "empty" pg.

how about rewriting it like:
```c++
auto existing_watch = crimson::osd::Watch::create(
  obc, winfo, src.second, pg);
existing_watch->disconnect();
assert(!obc->obs.oi.watchers.contains(src));
obc->watchers.emplace(src, std::move(existing_watch));
```

Reply:
Yeah, will change it:-)